### PR TITLE
OIL-89: A sandbox for OIL

### PR DESCRIPTION
**Pull request by ltparis2018**
- Demos for timeout, cookie data and reset
- CPC page integration demo
- Demos for __cmp() calls
- Tag soft blocking demo

Note: Demo for configuration manager is invocable, but does not work yet.

### CODE REVIEW

File: etc/webpack.common.js [hunk 1]
```diff
@@ -135,14 +134,10 @@ let config = {
...
-      /* Raw loader support for *.html
-       * Returns file content as string
-       *
-       * See: https://github.com/webpack/raw-loader
-       */
+      //process sandbox HTML Page hierarchy
       {
-        test: /\.html$/,
-        loader: 'raw-loader'
+        test: /sandbox\/.*\.html$/,
```
Comment (CanardSauvage) [inline etc/webpack.common.js]: Only the html files in the sandbox folder? I didn't quite get how this works

File: src/scripts/userview/userview_modal.js [hunk 2]
```diff
@@ -46,6 +46,7 @@ function startTimeOut() {
     hasRunningTimeout = setTimeout(function () {
       removeOilWrapperFromDOM();
       sendEventToHostSite(EVENT_NAME_TIMEOUT);
+      hasRunningTimeout = undefined;
```
Comment (CanardSauvage) [inline src/scripts/userview/userview_modal.js]: Was this a bug?

Comment (ltparis2018) [reply src/scripts/userview/userview_modal.js]: Yes. setTimeout() function returns the ID value of the timer that is set. If timer runs out 'hasRunningTimeout' variable was not reset and OIL thought the timer still runs. Therefore, the timeout could not be started again. This was not a problem for normal use of OIL layer but for the sandbox.

File: etc/webpack.common.js (hunk 1 above)
Comment (ltparis2018) [reply etc/webpack.common.js]: In our project the raw-loader was not really needed for HTML files. But for the sandbox files we need the html-loader now. We use the require() feature the html-loader provides.

**Pull request merged**